lib/spack/spack/test/concretization/core.py

```diff
@@ -468,6 +469,77 @@ def test_mixing_compilers_only_affects_subdag(self):
             assert x.satisfies("%clang") is not expected_gcc
             assert x.satisfies("%gcc") is expected_gcc
 
+    def test_disable_mixing_prevents_mixing(self):
+        with spack.config.override("concretizer", {"compiler_mixing": False}):
+            with pytest.raises(spack.error.UnsatisfiableSpecError):
+                spack.concretize.concretize_one("dt-diamond%clang ^dt-diamond-bottom%gcc")
+
+    def test_disable_mixing_override_by_package(self):
+        with spack.config.override("concretizer", {"compiler_mixing": ["dt-diamond-bottom"]}):
+            root = spack.concretize.concretize_one("dt-diamond%clang ^dt-diamond-bottom%gcc")
+            assert root.satisfies("%clang")
+            assert root["dt-diamond-bottom"].satisfies("%gcc")
+            assert root["dt-diamond-left"].satisfies("%clang")
+
+            with pytest.raises(spack.error.UnsatisfiableSpecError):
+                spack.concretize.concretize_one("dt-diamond%clang ^dt-diamond-left%gcc")
+
+    def test_disable_mixing_reuse(self, mutable_database):
+        # Install a spec
+        left = spack.concretize.concretize_one("dt-diamond-left %gcc")
+        PackageInstaller([left.package], fake=True, explicit=True).install()
```

Comment:
We can avoid the `PackageInstaller` and run just the concretizer, if we use lower level API calls. An example of a test like that is e.g. `test_version_weight_and_provenance` that uses `solver.driver.solve` and thus doesn't need a store, or a db.